src/taipy/core/taipy.py

```diff
@@ -463,3 +466,34 @@ def clean_all_entities() -> bool:
     _CycleManagerFactory._build_manager()._delete_all()
     _JobManagerFactory._build_manager()._delete_all()
     return True
+
+
+def export_scenario(
+    scenario_id: ScenarioId,
+    folder: str,
+):
+    """Export all related entities of a scenario to a folder.
+
+    Parameters:
+        scenario_id (ScenarioId): The id of the scenario to export.
+        folder (str): The folder to export. This folder must not be the storage folder.
+
+    """
+    if Path(folder).resolve() == Path(Config.global_config.storage_folder).resolve():
+        raise InvalidExportPath("The export folder must not be the storage folder.")
+
+    manager = _ScenarioManagerFactory._build_manager()
+    scenario = manager._get(scenario_id)
+    entity_ids = manager._get_owned_entity_ids(scenario)._union_all  # type: ignore
+    entity_ids.add(scenario_id)
+    entity_ids.add(scenario.cycle.id)
+
+    # Copy storage folder to the export folder
+    if Path(folder).exists():
+        shutil.rmtree(folder)
+    shutil.copytree(Config.global_config.storage_folder, folder, dirs_exist_ok=True)
+
+    # Remove all entities that are not related to the scenario
+    for f in Path(folder).rglob("*"):
+        if f.is_file() and f.stem not in entity_ids:
+            f.unlink()
```

Comment:
1/ This only works with Filesystem storage. The logic of getting all the entity ids is good. But we should move the logic of extracting the entities (from the ids) to the repositories. The purpose is to implement the same for other storage systems (mongo, SQL). 

2/ I believe this implementation will be problematic when the user has many scenarios and entities. It will copy everything which can take a while and remove the major part of the files. 
Since you already have the list of entities, it should be straightforward to directly copy the files of the entities related to the scenario.

Reply:
Yes, so I am thinking of adding a method in manager class to retrieve the path of an entity by id. About the Mongo and SQL repo, we should export them into another database, am I correct?

Reply:
I would put this method in the FileSystemRepository. not in the manager.
The mongo and SQL repo should also export the data into the same folder.